src/server/game/Entities/Player/Player.cpp

```diff
@@ -18672,31 +18672,31 @@ bool Player::Satisfy(AccessRequirement const* ar, uint32 target_map, bool report
 
 bool Player::CheckInstanceLoginValid(Map* map)
 {
-    if (!map->IsDungeon() || IsGameMaster())
+    if (!map->IsDungeon() || IsInstanceLoginGameMasterException())
         return true;
 
     if (map->IsRaid())
     {
         // cannot be in raid instance without a group
         if (!GetGroup())
-            return IsInstanceLoginGameMasterException();
+            return false;
     }
     else
     {
         // cannot be in normal instance without a group and more players than 1 in instance
         if (!GetGroup() && map->GetPlayersCountExceptGMs() > 1)
-            return IsInstanceLoginGameMasterException();
+            return false;
     }
 
     // do checks for satisfy accessreqs, instance full, encounter in progress (raid), perm bind group != perm bind player
-    return sMapMgr->CanPlayerEnter(map->GetId(), this, true) || IsInstanceLoginGameMasterException();
+    return sMapMgr->CanPlayerEnter(map->GetId(), this, true);
 }
 
 bool Player::IsInstanceLoginGameMasterException() const
 {
-    if (CanBeGameMaster())
+    if (IsGameMaster() || CanBeGameMaster())
```

Comment:
With CanBeGameMaster you don't need to check IsGameMaster - CanBeGameMaster() checks if you have the rbac permission to use .gm on, while IsGameMaster() returns true if you used .gm on